mayastor/src/replicas/rebuild/rebuild_impl.rs

```diff
@@ -334,14 +337,39 @@ impl std::fmt::Display for RebuildOperation {
 }
 
 impl ClientOperations for RebuildJob {
-    fn stats(&self) -> Option<RebuildStats> {
+    fn stats(&self) -> RebuildStats {
+        let blocks_total = self.end - self.start;
+
+        // segment size may not be aligned to the total size
+        let blocks_recovered = std::cmp::min(
+            self.tasks.segments_done * self.segment_size_blks,
+            blocks_total,
+        );
+
+        let progress = (blocks_recovered * 100) / blocks_total;
+
         info!(
-            "State: {:#}, Src: {}, Dst: {}, start: {}, end: {}, next: {}, block: {}",
-            self.state(), self.source, self.destination,
-            self.start, self.end, self.next, self.block_size
+            "State: {}, Src: {}, Dst: {}, start: {}, end: {}, next: {}, \
+             block: {}, segment_sz: {}, recovered_blks: {}, progress: {}%",
```

Comment:
block_size has been renamed to block. I think block_size was clearer.

Reply:
oo don't know why I did that... fixed